site/content/3.12/release-notes/version-3.12/incompatible-changes-in-3-12.md

```diff
@@ -6,6 +6,166 @@ description: >-
+   empty list of affected indexes and an according message.
+   
+   ```json
+   {
+     "error": false,
+     "code": 200,
+     "result": {
+       "affected": [],
+       "error": false,
+       "errorCode": 0,
+       "errorMessage": "all good with sorting order"
+     }
+   }
+   ```
+   
+   If this is the case, create a backup and upgrade your deployment to the
+   latest bugfix version with the same major and minor version (e.g. from 3.11.x
+   to at least 3.11.11 or from 3.12.x to at least 3.12.2). Complete the procedure
+   by calling the `PUT /_admin/cluster/vpackSortMigration/migrate` endpoint to
+   mark the deployment as having the correct sorting order. This requires
+   [superuser permissions](../../develop/http-api/authentication.md#jwt-superuser-tokens)
+   unless authentication is disabled.
+
+   Make sure that no problematic values are written to or removed from an index
+   between checking for affected indexes and completing the procedure. You may
+   put the deployment into [read-only mode](../../develop/http-api/administration.md#set-the-server-mode-to-read-only-or-default)
+   temporarily.
+
+   ```shell
+   curl -H "Authorization: bearer <superuser-token>" -XPUT http://localhost:8529/_admin/cluster/vpackSortMigration/migrate
+   ```
+
+   ```json
+   {
+     "error": false,
+     "code": 200,
+     "result": {
+       "error": false,
+       "errorCode": 0,
+       "errorMessage": "VPack sorting migration done."
+     }
+   }
+   ```
+
+5. If affected indexes are found, the check result looks similar to this:
+
+   ```json
+   {
+     "error": false,
+     "code": 200,
+     "result": {
+       "affected": [
+         {
+           "database": "_system",
+           "collection": "coll",
+           "indexId": 195,
+           "indexName": "idx_1806192152446763008"
+         }
+       ],
+       "error": true,
+       "errorCode": 1242,
+       "errorMessage": "some indexes have legacy sorted keys"
+     }
+   }
+   ```
+
+   If this is the case, the next steps depend on the deployment type.
+
+   - **Single server**: Create a full dump with [arangodump](../../components/tools/arangodump/_index.md),
+     using the `--all-databases` and `--include-system-collections` startup options
+     and a user account with administrate access to the `_system` database and
+     at least read access to all other databases to ensure all data including
+     the `_users` system collection are dumped.
+     
+     Restore the dump to a new single server using at least v3.11.11 or v3.12.2.
+     You need to use a new database directory.
+
+   - **Cluster**: Replace the DB-Server nodes until they all run at least
+     v3.11.11 or v3.12.2. Syncing new nodes writes the data in the correct order.
+     This deployment mode and approach avoids downtimes.
+
+     For each DB-Server, add a new DB-Server node to the cluster. Wait until all
+     new DB-Servers are in sync, then clean out the old DB-Server nodes.
```

Comment:
The cluster admin endpoints cleanOutServer, resignLeadership, etc. are currently not documented, and failing over/removing nodes/machines could be described better.